platform/mbed_application.c

```diff
@@ -170,7 +173,7 @@ __asm static void start_new_application(void *sp, void *pc)
 void start_new_application(void *sp, void *pc)
 {
     __asm volatile(
-        "movw   r2, #0      \n" // Fail to compile "mov r2, #0" with ARMC6. Replace with MOVW.
+        "movs   r2, #0      \n" 
```

Comment:
Glad that worked. (It's so fiddly getting inline assembler to go smoothly across all toolchains - especially as GCC refuses to use unified syntax for Thumb-1 devices. Actually a bit surprised it is accepting MOVS...) 

The MOVT comment is no longer relevant with the MOVW gone.